packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -665,15 +665,22 @@ export class Project extends ProjectBase {
       encryptionKey: props.encryptionKey && props.encryptionKey.keyArn,
       badgeEnabled: props.badge,
       cache,
-      name: props.projectName,
+      name: new cdk.Token(() => this.tryPhysicalName).toString(),
```

Comment:
Seems like this will be a common pattern. Make `tryPhysicalName` already construct the Token?

Reply:
Also, I find the name `tryPhysicalName` confusing.

I would expect that `try` is followed by a verb, like `tryFind`, `tryEatCookie`, not a noun, like `tryBoat`.

Reply:
Done.